packages/api/src/module/write/email-confirmation/presentation/rest/email-confirmation.rest-controller.ts

```diff
@@ -0,0 +1,35 @@
+import { Body, Controller, HttpCode, Inject, Post, UseGuards } from '@nestjs/common';
+import { CommandBus } from '@nestjs/cqrs';
+
+import { JwtUserId } from '@/crud/auth/jwt/jwt-user-id.decorator';
+import {
+  RequestEmailConfirmationApplicationCommand,
+  requestEmailConfirmationCommand,
+} from '@/module/commands/request-email-conformation';
+import { UserId } from '@/shared/domain.types';
+import { JwtAuthGuard } from '@/shared/guards/jwt-auth.guard';
+import { ApplicationCommandFactory } from '@/write/shared/application/application-command.factory';
+import { ID_GENERATOR, IdGenerator } from '@/write/shared/application/id-generator';
+
+@Controller('email-confirmation')
+export class EmailConfirmationRestController {
+  constructor(
+    private readonly commandBus: CommandBus,
+    private readonly commandFactory: ApplicationCommandFactory,
+    @Inject(ID_GENERATOR)
+    private readonly idGenerator: IdGenerator,
```

Comment:
You don't need idGenerator, check out commandFactory implementation :)

Reply:
ApplicationCommandFactory has injected IdGenerator in the same way as controller and its necessary for confirmationToken 🤔

Reply:
IdGenerator is passed as function argument in applicationCommand method.
It was done to do not force you to inject IdGenerator inside controller. You can write: 
```ts
const command = this.commandFactory.applicationCommand((idGenerator) => ({
      class: RequestEmailConfirmationApplicationCommand,
      ...requestEmailConfirmationCommand({ userId, confirmationToken: idGenerator.generate(), confirmationFor }),
    }));
```